allennlp/tests/data/dataset_readers/universal_dependencies_dataset_reader_test.py

```diff
@@ -46,3 +46,19 @@ def test_read_from_file(self):
         assert fields["head_tags"].labels == ['root', 'punct', 'case', 'compound', 'root', 'case',
                                               'nmod', 'flat', 'flat', 'punct']
         assert fields["head_indices"].labels == [0, 4, 4, 4, 0, 6, 4, 6, 6, 4]
+
+        # This instance tests specifically for filtering of implicit references.
+        # The original sentence is:
+        # "Over 300 Iraqis are reported dead and 500 [reported] wounded in Fallujah alone."
+        # But the second "reported" has no head, because it is implicitly referenced.
```

Comment:
It's not that it has no head, it's that it isn't actually there, it's been elided.